### PR TITLE
Live execution memory consumption test framework

### PR DESCRIPTION
**Pull request by junmendoza**
Testing strategy for live execution memory consumption
Runtime Diagnostics implements functionality to peek into the VM runtime core - this will be extended 

Includes a simple workflow test that checks instructions incrementing in the instr cache despite only modifying values. This test currently fails. 

The test will only pass if value modifications no longer generate new instructions

@ke-yu @lukechurch PTAL

### CODE REVIEW

**Comment by junmendoza** [comment]
test/Engine/ProtoTest/FFITests/CSFFIDispose.cs is having diff issues as its a large change where I replace:
CreateSubTreeFromCode to ProtoTestFx.TD.TestFrameWork.CreateSubTreeFromCode

This is similar to the other files

File: src/Engine/ProtoCore/Diagnostics/RuntimeDiagnostics.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+using ProtoCore.AST;
+using ProtoCore.AST.AssociativeAST;
+using ProtoCore.BuildData;
+using ProtoCore.CodeModel;
+using ProtoCore.DebugServices;
+using ProtoCore.DSASM;
+using ProtoCore.Lang;
+using ProtoCore.Lang.Replication;
+using ProtoCore.Runtime;
+using ProtoCore.Utils;
```
Comment (ke-yu) [inline src/Engine/ProtoCore/Diagnostics/RuntimeDiagnostics.cs]: Do we really need to use all these namespaces? :-)

File: test/Engine/ProtoTest/LiveRunnerTests/MemoryConsumptionTests.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            base.Setup();
+            liverunner = new ProtoScript.Runners.LiveRunner(); 
+            runtimeDiagnostics = new ProtoCore.Diagnostics.Runtime(liverunner.RuntimeCore);
+        }
+
+        public override void TearDown()
+        {
+            base.TearDown();
+            liverunner.Dispose();
```
Comment (ke-yu) [inline test/Engine/ProtoTest/LiveRunnerTests/MemoryConsumptionTests.cs]: Although it is just for testing, but I think for initialization, initialize base's stuff firstly. For disposing, in reverse order.

**Comment by ke-yu** [comment]
LGTM.

File: src/Engine/ProtoCore/Diagnostics/RuntimeDiagnostics.cs (hunk 1 above)
Comment (junmendoza) [inline src/Engine/ProtoCore/Diagnostics/RuntimeDiagnostics.cs]: nah... removed most of it :)

File: test/Engine/ProtoTest/LiveRunnerTests/MemoryConsumptionTests.cs (hunk 2 above)
Comment (junmendoza) [inline test/Engine/ProtoTest/LiveRunnerTests/MemoryConsumptionTests.cs]: done

**Pull request merged**